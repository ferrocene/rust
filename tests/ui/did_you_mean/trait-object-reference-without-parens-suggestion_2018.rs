//@ edition:2018..

fn main() {
    let _: &dyn Copy + 'static; //~ ERROR ambiguous `+` in a type
    //~^ ERROR is not dyn compatible
    let _: &'static dyn Copy + 'static; //~ ERROR ambiguous `+` in a type
    //~^ ERROR is not dyn compatible
}
