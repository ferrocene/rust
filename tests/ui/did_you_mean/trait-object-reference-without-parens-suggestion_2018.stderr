error: ambiguous `+` in a type
  --> $DIR/trait-object-reference-without-parens-suggestion_2018.rs:4:13
   |
LL |     let _: &dyn Copy + 'static;
   |             ^^^^^^^^^^^^^^^^^^
   |
help: try adding parentheses
   |
LL |     let _: &(dyn Copy + 'static);
   |             +                  +

error: ambiguous `+` in a type
  --> $DIR/trait-object-reference-without-parens-suggestion_2018.rs:6:21
   |
LL |     let _: &'static dyn Copy + 'static;
   |                     ^^^^^^^^^^^^^^^^^^
   |
help: try adding parentheses
   |
LL |     let _: &'static (dyn Copy + 'static);
   |                     +                  +

error[E0038]: the trait `Copy` is not dyn compatible
  --> $DIR/trait-object-reference-without-parens-suggestion_2018.rs:4:17
   |
LL |     let _: &dyn Copy + 'static;
   |                 ^^^^ `Copy` is not dyn compatible
   |
   = note: the trait is not dyn compatible because it requires `Self: Sized`
   = note: for a trait to be dyn compatible it needs to allow building a vtable
           for more information, visit <https://doc.rust-lang.org/reference/items/traits.html#dyn-compatibility>

error[E0038]: the trait `Copy` is not dyn compatible
  --> $DIR/trait-object-reference-without-parens-suggestion_2018.rs:6:25
   |
LL |     let _: &'static dyn Copy + 'static;
   |                         ^^^^ `Copy` is not dyn compatible
   |
   = note: the trait is not dyn compatible because it requires `Self: Sized`
   = note: for a trait to be dyn compatible it needs to allow building a vtable
           for more information, visit <https://doc.rust-lang.org/reference/items/traits.html#dyn-compatibility>

error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0038`.
